design-proposals/status-for-nncp-not-matching-any-node.md

```diff
@@ -0,0 +1,192 @@
+# Status for NNCP not matching any node(s)
+
+## Summary
+
+This proposal is trying to figure out a way on how to handle the status response kubernetes-nmstate is reporting on
+`oc get nncp -oyaml` when the NNCP is not matching any node(s).
+
+## Motivation
+
+Users unfamiliar with the status reports of `oc get nncp -oyaml` might confuse the status report of `status.conditions[].type`
+(Available) and incorrectly assume that a NNCP has been properly applied to node(s) even though it is possibly not
+matching any node(s) `status.conditions[].message` (Policy does not match any node) and `status.conditions[].reason`
+(NoMatchingNode). Tickets like [these](https://issues.redhat.com/browse/OPNET-81) have already come up in the past and
+should therefore be addressed to provide a clear answer.
+
+### User Stories
+
+- As a cluster administrator, I want to be able to easily judge if an NNCP has successfully been rolled out by the status.
+
+### Goals
+
+- Clarify which `status.condition[]` list an applied NNCP not matching any node(s) should have.
+
+### Non-Goals
+
+- Changes to `status.condition[]` on anything else than applied NNCP.
+
+## Proposal
+
+### Solution 1: Add new status [completed/ignored]
+
+Adding a new status will indicate to the user that the NNCP is neither `progressing`, `degraded` nor `available`.
+That status could be called `completed` or `ignored`. `Completed` could indicate that a NNCP has applied but is not "running"
+therefore only `completed` and not `available`. `Ignored` could be used to indicate that the NNCP is valid therefore not
+`degraded` but is being ignored by all nodes
+for any reason (e.g. wrong node selectors).
```

Comment:
I am personally against introducing concept of `Completed` next to `Available` because applying NNCP is an oneshot action. So saying "NNCP [...] is not running" is fundamentally wrong, because NNCP as a construct cannot be running, it can either apply or not apply. Think about it as an event, not a process.

`Ignored` next to `Available` and `Degraded` is much nicer though.